### ISSUE TITLE
Can't craft a barn

### ISSUE THREAD

**Issue by webcrafter-fr** [Brian-Wuest/MC-Prefab#70]
I have all the items but it don't work.

**Comment by Brian-Wuest** [comment]
Please provide more information. "it don't work" doesn't help me help you.

1. What version of Minecraft and forge are you running?
2. What version of the mod are you running? (Make sure you are running the latest for the version of minecraft you are running.)
3. Are you running a specific modpack on CurseForge that I can use to test it out?
4. Did you check the configuration file to determine if the Barn recipe got disabled? (All recipes are enabled by default unless you or a modpack developer disabled them.)
5. Does the recipe show up in JEI?
6. If you are using Extra Utilities make sure you aren't using their version of Compressed Stone/Dirt.
7. Are other blueprints working? Such as the regular Chicken Coop?

If you are still having an issue after checking all of these things, please provide details of what you are and are not seeing in the game and provide a log file through GIST or PasteBin.

**Comment by webcrafter-fr** [comment]
Minecraft : 1.12.2
Forge : 14.23.1.2555
Mod version : 1.3.0.13

My mods : 
- Clockhub
- craftsaddles
- Fullscreenwindowed
- Hearthstonemod
- Inventory Tweaks
- MInecolonies
- NaturesCompass
- SignPost
- WearableBackpacks

All recipes are enable.

What is JEI ? I can generate the barn blueprint in the creative mod

I don't think my others mods are using a version of compressed stone / dirt.

For the moment I tried the start house and the nether portal. It works fine.

**Comment by webcrafter-fr** [comment]
Ok so I tried every structure craft and there is two recipes who don't work : barn and Aqua base

**Comment by Brian-Wuest** [comment]
Very interesting, next set of questions. 
1. You said the recipe doesn't work, you're talking about the crafting recipe? So you don't even get the item?
2. When you pulled the item from creative menu does the normal screen come up when right clicked? And if so, does the preview function work?
3. JEI is a mod called "Just Enough Items". It has functionality which allows you to see all items and blocks like the creative menu but also works in survival. It can also show you the recipies for the block/item and what recipes the block/item are used in.

**Comment by webcrafter-fr** [comment]
1. Yep, I place the items (x2 pallet of bricks, 1 compressed glowstone, 5 bundle of timber, 1 double compressed dirt) but my crafting table does't show me the result. The output stay empty.

2. All functions are working. It's just a crafting issue.

3. Ok :)

**Comment by Brian-Wuest** [comment]
Thank you, I will look at this when I get home.

**Comment by Brian-Wuest** [comment]
I got the aqua base and barn to work using the mods you specified above. Please check out this [imgur link](https://imgur.com/a/A4tZN).

Can you upload your prefab.cfg so I can see what your configuration is like?

**Comment by webcrafter-fr** [comment]
Yep : 

`# Configuration file

general {

    options {
        # Determines if the house item is added to player inventory when joining the world for the first time. Server configuration overrides client. [default: true]
        B:"Add House Item On New Player Join"=true

        # Determines if the moderate house item is provided to the player instead of the starting house. [default: false]
        B:"Add Moderate House on World Join"=false

        # When true this option causes the Crafting Table, Furnace and Chest to not be added when creating a house, regardless of options chosen. Server Configuration overrides client. [default: false]
        B:"Enable House Generation Restrictions"=false
        B:"Enable Loft House"=false
        B:"Enable Version Checking"=true

        # Determines if the mineshaft chest is included when building mineshafts for various structures. [default: true]
        B:"Include Mineshaft Chest"=true

        # Determines if the spawners for the Monster Masher building are included. Server configuration overrides client. [default: true]
        B:"Include Spawners in Monster Masher"=true

        # Determines if the Preview buttons in structure GUIs and other structure previews functions are enabled. Client side only. [default: true]
        B:"Include Structure Previews"=true

        # Determines the maximum size the starting house can be generated as. Server configuration overrides client. [range: 5 ~ 16, default: 16]
        I:"Maximum Starting House Size"=16

        ##########################################################################################################
        # chest contents
        #--------------------------------------------------------------------------------------------------------#
        # This category is to determine the contents of the chest created by the house item. When playing on a server, the server configuration is used.
        ##########################################################################################################

        "chest contents" {
            # Determines if Leather Armor is added the the chest when the house is created. [default: true]
            B:"Add Armor"=true

            # Determines if a Stone Axe is added the the chest when the house is created. [default: true]
            B:"Add Axe"=true

            # Determines if a stack of cobble is added the the chest when the house is created. [default: true]
            B:"Add Cobblestone"=true

            # Determines if seeds, potatoes and carros are added the the chest when the house is created. [default: true]
            B:"Add Crops"=true

            # Determines if a stack of dirt is added the the chest when the house is created. [default: true]
            B:"Add Dirt"=true

            # Determines if Bread is added the the chest when the house is created. [default: true]
            B:"Add Food"=true

            # Determines if a Stone Hoe is added the the chest when the house is created. [default: true]
            B:"Add Hoe"=true

            # Determines if a Stone Pickaxe is added the the chest when the house is created. [default: true]
            B:"Add Pickaxe"=true

            # Determines if a set of oak saplings are added the the chest when the house is created. [default: true]
            B:"Add Saplings"=true

            # Determines if a Stone Shovel is added the the chest when the house is created. [default: true]
            B:"Add Shovel"=true

            # Determines if a Stone Sword is added the the chest when the house is created. [default: true]
            B:"Add Sword"=true

            # Determines if a set of torches are added the the chest when the house is created. [default: true]
            B:"Add Torches"=true
        }

        ##########################################################################################################
        # recipes
        #--------------------------------------------------------------------------------------------------------#
        # This category determines if the recipes for the blocks/items in this are enabled
        ##########################################################################################################

        recipes {
            # Determines if the recipe(s) associated with the Advanced Chicken Coop are enabled. [default: true]
            B:"Advanced Chicken Coop"=true

            # Determines if the recipe(s) associated with the Advanced Horse Stable are enabled. [default: true]
            B:"Advanced Horse Stable"=true

            # Determines if the recipe(s) associated with the Advanced Warehouse are enabled. [default: true]
            B:"Advanced Warehouse"=true

            # Determines if the recipe(s) associated with the Andesite Slabs are enabled. [default: true]
            B:"Andesite Slabs"=true

            # Determines if the recipe(s) associated with the Andesite Stairs are enabled. [default: true]
            B:"Andesite Stairs"=true

            # Determines if the recipe(s) associated with the Aqua Base are enabled. [default: true]
            B:"Aqua Base"=true

            # Determines if the recipe(s) associated with the Barn are enabled. [default: true]
            B:Barn=true

            # Determines if the recipe(s) associated with the Bulldozer are enabled. [default: true]
            B:Bulldozer=true

            # Determines if the recipe(s) associated with the Bundle of Timber are enabled. [default: true]
            B:"Bundle of Timber"=true

            # Determines if the recipe(s) associated with the Chicken Coop are enabled. [default: true]
            B:"Chicken Coop"=true

            # Determines if the recipe(s) associated with the Compressed Chest are enabled. [default: true]
            B:"Compressed Chest"=true

            # Determines if the recipe(s) associated with the Compressed Dirt are enabled. [default: true]
            B:"Compressed Dirt"=true

            # Determines if the recipe(s) associated with the Compressed Glowstone are enabled. [default: true]
            B:"Compressed Glowstone"=true

            # Determines if the recipe(s) associated with the Compressed Obsidian are enabled. [default: true]
            B:"Compressed Obsidian"=true

            # Determines if the recipe(s) associated with the Compressed Stone are enabled. [default: true]
            B:"Compressed Stone"=true

            # Determines if the recipe(s) associated with the Defense Bunker are enabled. [default: true]
            B:"Defense Bunker"=true

            # Determines if the recipe(s) associated with the Diorite Slabs are enabled. [default: true]
            B:"Diorite Slabs"=true

            # Determines if the recipe(s) associated with the Diorite Stairs are enabled. [default: true]
            B:"Diorite Stairs"=true

            # Determines if the recipe(s) associated with the Ender Gateway are enabled. [default: true]
            B:"Ender Gateway"=true

            # Determines if the recipe(s) associated with the Fish Farm are enabled. [default: true]
            B:"Fish Farm"=true

            # Determines if the recipe(s) associated with the Glass Slabs are enabled. [default: true]
            B:"Glass Slabs"=true

            # Determines if the recipe(s) associated with the Glass Stairs are enabled. [default: true]
            B:"Glass Stairs"=true

            # Determines if the recipe(s) associated with the Granite Slabs are enabled. [default: true]
            B:"Granite Slabs"=true

            # Determines if the recipe(s) associated with the Granite Stairs are enabled. [default: true]
            B:"Granite Stairs"=true

            # Determines if the recipe(s) associated with the Grassy Plains are enabled. [default: true]
            B:"Grassy Plains"=true

            # Determines if the recipe(s) associated with the Green House are enabled. [default: true]
            B:"Green House"=true

            # Determines if the recipe(s) associated with the Horse Stable are enabled. [default: true]
            B:"Horse Stable"=true

            # Determines if the recipe(s) associated with the Instant Bridge are enabled. [default: true]
            B:"Instant Bridge"=true

            # Determines if the recipe(s) associated with the Machinery Tower are enabled. [default: true]
            B:"Machinery Tower"=true

            # Determines if the recipe(s) associated with the Magic Temple are enabled. [default: true]
            B:"Magic Temple"=true

            # Determines if the recipe(s) associated with the Mineshaft Entrance are enabled. [default: true]
            B:"Mineshaft Entrance"=true

            # Determines if the recipe(s) associated with the Moderate House are enabled. [default: true]
            B:"Moderate House"=true

            # Determines if the recipe(s) associated with the Monster Masher are enabled. [default: true]
            B:"Monster Masher"=true

            # Determines if the recipe(s) associated with the Nether Gate are enabled. [default: true]
            B:"Nether Gate"=true

            # Determines if the recipe(s) associated with the Paper Lantern are enabled. [default: true]
            B:"Paper Lantern"=true

            # Determines if the recipe(s) associated with the Phasic Block are enabled. [default: true]
            B:"Phasic Block"=true

            # Determines if the recipe(s) associated with the Pile of Bricks are enabled. [default: true]
            B:"Pile of Bricks"=true

            # Determines if the recipe(s) associated with the Produce Farm are enabled. [default: true]
            B:"Produce Farm"=true

            # Determines if the recipe(s) associated with the Smart Glass are enabled. [default: true]
            B:"Smart Glass"=true

            # Determines if the recipe(s) associated with the Starting House are enabled. [default: true]
            B:"Starting House"=true

            # Determines if the recipe(s) associated with the Tree Farm are enabled. [default: true]
            B:"Tree Farm"=true

            # Determines if the recipe(s) associated with the Villager Houses are enabled. [default: true]
            B:"Villager Houses"=true

            # Determines if the recipe(s) associated with the Warehouse are enabled. [default: true]
            B:Warehouse=true

            # Determines if the recipe(s) associated with the Warehouse Upgrade are enabled. [default: true]
            B:"Warehouse Upgrade"=true

            # Determines if the recipe(s) associated with the Watch Tower are enabled. [default: true]
            B:"Watch Tower"=true
        }

    }

}


`

**Comment by Brian-Wuest** [comment]
Hmm, everything seems to be in in order here and I copied it over to my configuration file and it works just fine. 

You are running single player right or are you running on a multiplayer server?

If you are running in single player then can you zip up all of your configs and put them in a place where I can download them? There could be some weird setting somewhere.

If you are running on a multiplayer server, the server configuration overrides whatever you have set locally.

**Comment by webcrafter-fr** [comment]
Single player. 

I uploaded my config files here : https://ufile.io/ep5i4

I didn't change anything in my config files :/ I just loaded my mods and played the game.

**Comment by Brian-Wuest** [comment]
Alright, this is getting frustrating. After getting your config files and running a general minecraft instance with the mod version you specified, I could not duplicate your issue. I also tried Creative and Survival modes and everything works fine. They show up in the crafting table and I can get the item without issue.

I have updated the [imgur link](https://imgur.com/a/A4tZN) with a screen shot of what it looks like with JEI showing the recipes.

At this point I am at a loss as to why you cannot craft these items.
